src/main/java/edu/byu/cs/autograder/Grader.java

```diff
@@ -228,26 +239,47 @@ private Submission saveResults(Rubric rubric, int numCommits) {
         );
 
         if (submission.passed()) {
-            sendToCanvas(submission);
+            sendToCanvas(submission, 1 - (numDaysLate * 0.1F));
         }
 
         submissionDao.insertSubmission(submission);
 
         return submission;
     }
 
-    private void sendToCanvas(Submission submission) {
+    private void sendToCanvas(Submission submission, float lateAdjustment) {
         UserDao userDao = DaoService.getUserDao();
         User user = userDao.getUser(netId);
 
         int userId = user.canvasUserId();
 
         int assignmentNum = PhaseUtils.getPhaseAssignmentNumber(phase);
 
-        float score = submission.score() * PhaseUtils.getTotalPoints(phase);
+        RubricConfig rubricConfig = DaoService.getRubricConfigDao().getRubricConfig(phase);
+        Map<String, Float> scores = new HashMap<>();
+        Map<String, String> comments = new HashMap<>();
+        if(rubricConfig.passoffTests() != null) {
+            String id = getCanvasRubricId(Rubric.RubricType.PASSOFF_TESTS);
+            Rubric.Results results = submission.rubric().passoffTests().results();
+            scores.put(id, results.score() * lateAdjustment);
+            comments.put(id, results.notes());
+        }
+        if(rubricConfig.unitTests() != null) {
+            String id = getCanvasRubricId(Rubric.RubricType.UNIT_TESTS);
+            Rubric.Results results = submission.rubric().unitTests().results();
+            scores.put(id, results.score() * lateAdjustment);
+            comments.put(id, results.notes());
+        }
+        if(rubricConfig.quality() != null) {
+            String id = getCanvasRubricId(Rubric.RubricType.QUALITY);
+            Rubric.Results results = submission.rubric().quality().results();
+            scores.put(id, results.score() * lateAdjustment);
+            comments.put(id, results.notes());
+        }
+
 
         try {
-            CanvasIntegration.submitGrade(userId, assignmentNum, score, submission.notes());
+            CanvasIntegration.submitGrade(userId, assignmentNum, scores, comments, submission.notes());
         } catch (CanvasException e) {
             LOGGER.error("Error submitting score for user " + submission.netId(), e);
```

Comment:
I know we weren't already doing this before your change, but I think we should throw an exception here instead so that the user would be notified that they didn't  get a grade in Canvas

Reply:
Instead of just logging it